modules/library/main/src/main/java/org/geotools/data/util/ScreenMap.java

```diff
@@ -134,17 +135,24 @@ public boolean checkAndSet(int x, int y) {
         int mask = 1 << offset;
 
         try {
-            if ((pixels[index] & mask) != 0) {
+            if ((getPixels()[index] & mask) != 0) {
                 return true;
             } else {
-                pixels[index] = pixels[index] | mask;
+                getPixels()[index] = getPixels()[index] | mask;
                 return false;
             }
         } catch (Exception e) {
             return true;
         }
     }
 
+    private int[] getPixels() {
+        if (pixels == null) {
+            pixels = new int[arraySize];
```

Comment:
might want to check arraySize isn't too big here and that there is no OOM issue

Reply:
Oh sorry, these changes got in by mistake, generating vector tiles allocates a 4000x4000 screenarea now, often for nothing, so I was trying to delay it, but this experimental code should be cleaned up. I have to remove it from this PR, it's unrelated

Reply:
Actually, found a better way to collect min and max in a single pass, will update the PR.